src/software/ai/hl/stp/action/intercept_ball_action.cpp

```diff
@@ -73,9 +73,14 @@ void InterceptBallAction::calculateNextIntent(IntentCoroutine::push_type& yield)
         bool point_in_front_of_ball =
             pointInFrontVector(ball.position(), ball.velocity(), closest_point);
 
-        // We add 1e-6 to avoid division by 0 without affecting the result significantly
-        Duration ball_time_to_position = Duration::fromSeconds(
-            distance(closest_point, ball.position()) / (ball.velocity().length() + 1e-6));
+        Duration ball_time_to_position;
+
+        if (ball.velocity().length() != 0)
+        {
+            ball_time_to_position = Duration::fromSeconds(
+                distance(closest_point, ball.position()) / (ball.velocity().length()));
+        }
+
```

Comment:
`if (ball.velocity().length() == 0)` then `ball_time_to_position > robot_time_to_pos` must be true, so it should be integrated into the following code

```
        if (point_in_front_of_ball && (ball_time_to_position > robot_time_to_pos))
        {
            intercept_pos = closest_point;
        }
```

Reply:
Done.

Reply:
IMO this is about as arbitrary as just adding an EPSILON to the denominator. `1e15` here is just as "arbitrary" as the EPSILON value we use. I think adding an EPSILON is a cleaner solution

Reply:
Changed so if the check is true, add an EPSILON.